FirebaseAuth/Sources/Public/FirebaseAuth/FIRAuth.h

```diff
@@ -17,6 +17,7 @@
 #import <AvailabilityMacros.h>
 #import <Foundation/Foundation.h>
 
+#import <AuthenticationServices/AuthenticationServices.h>
```

Comment:
@paulb777, should types from this framework instead be forward declared within the header?

Reply:
Yes, and wrapped in `#if TARGET_OS_IOS`

Reply:
can you elaborate on the forward declaration part?

another question on TARGET_OS_IOS, these APIs will run on mac os, ios and tv os (but not watch OS).
I was thinking using
`if TARGET_OS_IOS || TARGET_OS_TV || TARGET_OS_MAC`.  However going thru this [doc](https://github.com/apple/swift-corelibs-foundation/blob/main/CoreFoundation/Base.subproj/SwiftRuntime/TargetConditionals.h#L52C9-L52C22), I have the assumption that TARGET_OS_MAC will include all IOS, TVOS and WATCHOS, can you suggest?

Reply:
> can you elaborate on the forward declaration part?

So line 20 should be moved to the implementation file. And in its place should be forward declarations for the types used from the `AuthenticationServices` module. The diff should look like:
```suggestion
@class ASAuthorizationPlatformPublicKeyCredentialAssertion;
@class ASAuthorizationPlatformPublicKeyCredentialAssertionRequest;
```

--- 

> another question on TARGET_OS_IOS, these APIs will run on mac os, ios and tv os (but not watch OS).
I was thinking using
if TARGET_OS_IOS || TARGET_OS_TV || TARGET_OS_MAC. However going thru this [doc](https://github.com/apple/swift-corelibs-foundation/blob/main/CoreFoundation/Base.subproj/SwiftRuntime/TargetConditionals.h#L52C9-L52C22), I have the assumption that TARGET_OS_MAC will include all IOS, TVOS and WATCHOS, can you suggest?

`TARGET_OS_MAC` might be too broad. Something like the following should specifically get all platforms except watchOS:

```objc
#if TARGET_OS_IOS || TARGET_OS_TV || TARGET_OS_OSX || TARGET_OS_MACCATALYST

#endif
```

Reply:
Thank you Nick.  After trying the suggestion we synced offline, it seems importing in the .m file still cause me compiler errors.  Can you help me take a look where I might did wrong? (CI is also failing for the same issue)